src/main/java/com/hotels/beeju/ThriftHiveMetaStoreJUnitRule.java

```diff
@@ -50,33 +52,45 @@ public ThriftHiveMetaStoreJUnitRule(String databaseName) {
    * Create a Thrift Hive Metastore service with a pre-created database using the provided name and configuration.
    *
    * @param databaseName Database name.
-   * @param preConfiguration Hive configuration properties that will be set prior to BeeJU potentially overriding these with its defaults.
+   * @param preConfiguration Hive configuration properties that will be set prior to BeeJU potentially overriding these
+   *          with its defaults.
    */
   public ThriftHiveMetaStoreJUnitRule(String databaseName, Map<String, String> preConfiguration) {
     super(databaseName, preConfiguration);
   }
-  
+
   /**
    * Create a Thrift Hive Metastore service with a pre-created database using the provided name and configuration.
    *
    * @param databaseName Database name.
-   * @param preConfiguration Hive configuration properties that will be set prior to BeeJU potentially overriding these with its defaults.
+   * @param preConfiguration Hive configuration properties that will be set prior to BeeJU potentially overriding these
+   *          with its defaults.
    * @param postConfiguration Hive configuration properties that will be set to override BeeJU's defaults.
    */
-  public ThriftHiveMetaStoreJUnitRule(String databaseName, Map<String, String> preConfiguration, Map<String, String> postConfiguration) {
+  public ThriftHiveMetaStoreJUnitRule(
+      String databaseName,
+      Map<String, String> preConfiguration,
+      Map<String, String> postConfiguration) {
     super(databaseName, preConfiguration, postConfiguration);
   }
 
   @Override
-  protected void before() throws Throwable {
-    thriftHiveMetaStoreCore.initialise();
-    super.before();
+  protected void starting(Description description) {
+    try {
+      thriftHiveMetaStoreCore.initialise();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    super.starting(description);
```

Comment:
Can you check if it still works (and if it makes sense) to move the call to `super` before the initialisation?

Reply:
If that's the case, it should be done in the `Extension` too.

Reply:
Yup, that seems to work and is how you did in the `hive-3.x` branch right? So might as well do that here now too.

Reply:
Actually, this change broke things in downstream projects with various connection failure issues so I've put it back to how I had it. Try this order in the `hive-3.x` branch and see if it's OK?